Realm/RLMObjectSchema.mm

```diff
@@ -188,6 +199,24 @@ + (NSArray *)propertiesForClass:(Class)objectClass isSwift:(bool)isSwiftClass {
         }
     }
 
+    if (NSArray *optionalProperties = [objectUtil getOptionalPropertyNames:swiftObjectInstance]) {
```

Comment:
it essentially is, through the different `objectUtil` classes. I prefer this approach to switching explicitly on `isSwift`